src/lua/luaclass.cpp

```diff
@@ -55,10 +55,16 @@ bool luaClass_c::getBool(const char *name) {
 
 /* functions to evaluate lua code */
 int luaClass_c::doFile(const char *fname) {
-  luaL_loadfile(L, fname) || lua_pcall(L, 0, 0, 0);
```

Comment:
I do not understand the reason to call lua_pcall.  Is this to extrract potential errors from the previous call?

Reply:
to add: Shouldn't the new code be 
```
  int ret = luaL_loadfile(L, fname);
  if (ret == 0) {
    return lua_pcall(L, 0, 0, 0);
  }
  return ret;
```

Reply:
This one is rather odd.  So the function returns an int, but has no return statement, which if I understand the semantics correctly means the "truthy" output of void, which is zero in integer space.  As for the Lua calls, the load file prepares the Lua VM for operation, loading and registering functions, etc.  The "pcall" actually executes the Lua VM.  Honestly don't care one way or another which return value to use.

Reply:
> So the function returns an int, but has no return statement, which if I understand the semantics correctly

According to the c++ standard, this is undefined behavior.  Therefore, I would return the lua return value.